cap/modules/deposit/validators.py

```diff
@@ -24,21 +24,49 @@
 """Deposit validators."""
 
 import os
-
+from flask_login import current_user
+from flask_principal import RoleNeed
+from invenio_access.permissions import Permission
 from jsonschema import Draft4Validator
 from jsonschema.validators import extend
+from jsonschema.exceptions import ValidationError
 
 from cap.modules.experiments.validators import (validate_cms_trigger,
                                                 validate_das_path,
                                                 validate_unique_cadi)
 
 deposit_validators = dict(Draft4Validator.VALIDATORS)
 
+
+# Validate for schema field permissions
```

Comment:
(1) lets move this method to another file (e.g. `cap/modules/deposit/validators/user.py`)
(2) lets make it more efficient: stop checking for permissions if allowed user or role is found